federation/pkg/kubefed/init/init.go

```diff
@@ -662,6 +669,7 @@ func createPVC(clientset client.Interface, namespace, svcName, federationName, e
 					api.ResourceStorage: capacity,
 				},
 			},
+			StorageClassName: storageClassName,
```

Comment:
I'm not sure what your concern is.  Isn't assigning a field a value of nil effectively the same as not setting the field?